caddy-gen/caddy-gen.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import yaml
+import logging
+import argparse
+import dataclasses as dc
+from config import BASE, LUG_ADDR, ROOT_DIR
+
+DESC = 'A simple Caddyfile generator for siyuan.'
+INDENT_CNT = 4
+
+
+@dc.dataclass
+class Node:
+    name: str
+    children: list['Node'] = dc.field(default_factory=list)
+
+    def __str__(self, level: int = 0):
+        if self.name == '':
+            return '\n\n'.join(
+                [child.__str__(level=level) for child in self.children])
+        elif len(self.children) == 0:
+            return ' ' * (level * INDENT_CNT) + self.name
+        else:
+            children_str = '\n'.join(
+                [child.__str__(level=level + 1) for child in self.children])
+            return ' ' * (level * INDENT_CNT) + self.name + ' {\n' + \
+                   children_str + '\n' + \
+                   ' ' * (level * INDENT_CNT) + '}'
+
+
+def https_redir() -> list[Node]:
+    return [
+        Node(f'http://{BASE}', [
+            Node('redir / https://{host}{uri} 301')
+        ])
+    ]
+
+
+def repo_redir(repo: dict) -> list[Node]:
+    return [Node(f'redir /{repo["name"]} /{repo["name"]}/ 301')]
```

Comment:
Seems that this rule is not applied to `Caddyfile.example`. Would you please update the example file?